ts/features/bonus/siciliaVola/store/reducers/voucherRequest.ts

```diff
@@ -0,0 +1,142 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import { getType } from "typesafe-actions";
+import {
+  Company,
+  Hospital,
+  University,
+  VoucherRequest
+} from "../../types/SvVoucherRequest";
+import {
+  FlightsDate,
+  svGenerateVoucherSelectCategory,
+  svGenerateVoucherSelectCompany,
+  svGenerateVoucherSelectFlightsDate,
+  svGenerateVoucherSelectHospital,
+  svGenerateVoucherSelectUniversity,
+  svGenerateVoucherStart,
+  svGenerateVoucherUnderThresholdIncome
+} from "../actions/voucherGeneration";
+import { Action } from "../../../../../store/actions/types";
+
+export type VoucherRequestState = Option<VoucherRequest>;
+const INITIAL_STATE: VoucherRequestState = none;
+
+/**
+ * Utils needed to update only the under threshold value
+ * @param state
+ * @param underThresholdIncome
+ */
+const updateUnderThresholdIncome = (
+  state: VoucherRequestState,
+  underThresholdIncome: boolean
+): VoucherRequestState =>
+  state.fold(state, vR => {
+    if (vR.category === "worker" || vR.category === "sick") {
+      return some({ ...vR, underThresholdIncome });
+    }
+    return state;
+  });
+
+/**
+ * Utils needed to update only the university value
+ * @param state
+ * @param university
+ */
+const updateUniversity = (
+  state: VoucherRequestState,
+  university: University
+): VoucherRequestState =>
+  state.fold(state, vR => {
+    if (vR.category === "student") {
+      return some({ ...vR, university });
+    }
+    return state;
+  });
+
+/**
+ * Utils needed to update only the company value
+ * @param state
+ * @param company
+ */
+const updateCompany = (
+  state: VoucherRequestState,
+  company: Company
+): VoucherRequestState =>
+  state.fold(state, vR => {
+    if (vR.category === "worker") {
+      return some({ ...vR, company });
+    }
+    return state;
+  });
+
+/**
+ * Utils needed to update only the hospital value
+ * @param state
+ * @param hospital
+ */
+const updateHospital = (
+  state: VoucherRequestState,
+  hospital: Hospital
+): VoucherRequestState =>
+  state.fold(state, vR => {
+    if (vR.category === "sick") {
+      return some({ ...vR, hospital });
+    }
+    return state;
+  });
+
+/**
+ * Utils needed to update only the flights date values
+ * @param state
+ * @param flightsDate
+ */
+const updateFlightsDate = (
+  state: VoucherRequestState,
+  flightsDate: FlightsDate
+): VoucherRequestState =>
+  state.fold(state, vR => {
+    if (
+      vR.category === "student" ||
+      vR.category === "worker" ||
+      vR.category === "sick"
+    ) {
```

Comment:
This check should be removed because it is common to every type.

Reply:
Updated in [51af9d6](https://github.com/pagopa/io-app/pull/3170/commits/51af9d6d69116f1e07bff54a7b22a21c57cd8aaa).